components/scroll-wrapper/scroll-wrapper.js

```diff
@@ -219,6 +283,58 @@ class ScrollWrapper extends RtlMixin(LitElement) {
 		this.scrollDistance(scrollDistance, true);
 	}
 
+	_synchronizeScroll(e) {
+		if (this._syncDriver && e.target !== this._syncDriver) return;
+		if (this._syncDriverTimeout) clearTimeout(this._syncDriverTimeout);
+
+		this._syncDriver = e.target;
+		this._allScrollers.forEach(element => {
+			if (element && element !== e.target) element.scrollLeft = e.target.scrollLeft;
+		});
+		this._syncDriverTimeout = setTimeout(() => this._syncDriver = null, 100);
+	}
+
+	_updateScrollTargets() {
+		this._disconnectAll();
+
+		if (this._printMode) return;
+
+		this._baseContainer = this.shadowRoot.querySelector('.d2l-scroll-wrapper-container');
+		this._container = this.customScrollers?.primary || this._baseContainer;
+		this._secondaryScrollers = this.customScrollers?.secondary || [];
+		if (this._secondaryScrollers.length === undefined) this._secondaryScrollers = [ this._secondaryScrollers ];
+		this._allScrollers = [ this._container, ...this._secondaryScrollers ];
+
+		if (this._container) {
+			this._container.classList.add('d2l-scroll-wrapper-focus');
+			const styleRoot = getStyleSheetInsertionPoint(this._container);
+			if (styleRoot && 'adoptedStyleSheets' in styleRoot) {
```

Comment:
This is what we need to do to apply the focus styles into the right DOM scope for the container. The approach I took was to essentially use `querySelector` as I travel up the DOM to find the correct document fragment in which to insert the CSS.

Browser support for constructible stylesheets is OK, and I've tested this in older browsers and it doesn't blow up. Instead, it ends up using the browser's default focus outline -- which looks very similar to ours.

Just using the built-in focus ring is an option here for us... it would certainly be a lot easier! @dbatiste FYI in case you have opinions as the existing focus styles were added for WCAG 2.2.

Reply:
Awesome, thanks Dave!

I just remembered I wanted to add a visual diff for the split scrollers, with something focusable in a "secondary scroller" that causes it to scroll, to make sure the primary scroller properly matches it. I'll try to add that later today.